pkg/build/generator/generator.go

```diff
@@ -211,6 +211,9 @@ func (g *BuildGenerator) checkLastVersion(bc *buildapi.BuildConfig, lastVersion
 // updateImageTriggers sets the LastTriggeredImageID on all the ImageChangeTriggers on the BuildConfig and
 // updates the From reference of the strategy if the strategy uses an ImageStream or ImageStreamTag reference
 func (g *BuildGenerator) updateImageTriggers(ctx kapi.Context, bc *buildapi.BuildConfig, from, triggeredBy *kapi.ObjectReference) error {
+	if triggeredBy == nil {
```

Comment:
@mfojtik please look at https://github.com/openshift/origin/pull/3748/files#diff-82f54c5f4bf72a64f869a69e557df8cbL121.
Without knowing anything about how it *should* work I can tell that we removed the checking in that particular change. As to what is the intended semantics, I need to wait for other people to get up so that I can ask them :-)